### PR TITLE
Fix #3152 Restrict some teams from being displayed during sign up process

### LINKED ISSUE #3152: Restrict some teams from being displayed during sign up process
**Feature**
#2950

**Describe the task**
Restrict some teams from being displayed during sign up process
Introduce a metadata field such as 'joinable', which allow the UI to filter and show the joinable teams during user sign up process. 
This enables a subset of teams within open metadata to be joinable by anyone.

cc: @harshach

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #3152 

### Describe your changes :

I worked on issue #3152 

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t, @darth-coder00 @vivekratnavel @ShaileshParmar11

### CODE REVIEW

**Comment by Sachin-chaurasiya** [comment]
Blocked by #3612

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx [hunk 1]
```diff
@@ -629,7 +644,30 @@ const TeamsPage = () => {
...
+                                id="join-team"
+                                onClick={() =>
+                                  onTeamSettingUpdate({
+                                    ...(currentTeam as Team),
+                                    isJoinable: !currentTeam?.isJoinable,
+                                  })
+                                }>
+                                <div className="switch" />
+                              </div>
+                              <label htmlFor="join-team">Opne to join</label>
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx:668]:
```suggestion
                              <label htmlFor="join-team">Open to join</label>
```

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx [hunk 2]
```diff
@@ -565,10 +565,18 @@ const TeamsPage = () => {
     return uniqueList;
   };
 
-  const handleUpdateOwner = (owner: Team['owner']) => {
+  const handleUpdateOwner = (
+    owner: Team['owner'],
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _tier = '',
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx:571]: Do we need this argument?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx:571]: @vivekratnavel , Yes, we pass callback method to manage component and tier is positional argument in that.

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**